### PR TITLE
Arm support

### PR DESCRIPTION
**Pull request by dillstead**
Add ARM support

### CODE REVIEW

File: src/compiler.c
Comment (nevakrien) [inline src/compiler.c:1]: This one is such a beauty really puts mine to shame with how short it is. great work.
the only thing you could improve is maybe add comments but honestly this is not a priority its completely fine to make assembly that's without comments.

File: src/io.c
Comment (nevakrien) [inline src/io.c:1]: So i looked at the 2 new functions and there is an issue with how u r setting the uninilized values.... doing that memset on the entire tape is potentially prohibitively expensive. I do see the value so it should be under a feature flag with #ifdef

**Comment by nevakrien** [comment]
if you prefer I can do cleanup here myself but it might take me a hot minute since I would need to set an arm emulator on my machine.

**Comment by nevakrien** [comment]
I am seeing a lack of tests...  I bet you had a few for when you developed so put them as an optional feature flag in make.
If you didn't have any tests then we need to at least run the existing tests on arm with my interpreter and your compiler and see that the output match.

**Comment by nevakrien** [comment]
I have made a feature branch where I am adressing the changes. I am working on making tests work on cross compilation to make life easier. would probably need to massage that compiler idea u made I think i may end up just making a new command but we will see. 

double checking did you make this as a cross compiler or did you compile nativly?

File: src/compiler.c
Comment (dillstead) [reply src/compiler.c:1]: Sure, I'll add some comments.

File: src/io.c
Comment (dillstead) [reply src/io.c:1]: Which memset? Are you talking about this in ReadTapeEx?  I'm only marking the designated number of tape positions as uninitialized, I'm not marking the entire tape.
    for (int i = 0; i < metadata.left_index - metadata.left_limit; i++) *cur++ = Bit_Uninit;
That should probably be changed to a memset.  I'm not sure why I didn't do that in the first place now that I looked back at the code.

**Comment by dillstead** [comment]
> I have made a feature branch where I am adressing the changes. I am working on making tests work on cross compilation to make life easier. would probably need to massage that compiler idea u made I think i may end up just making a new command but we will see.
> 
> double checking did you make this as a cross compiler or did you compile nativly?

Yes, I tested a variety of test cases before I submitted the pull request on a native ARM environment (original Raspberry Pi).  The test cases I ran:

stay 
write 1, stay 
move left 
move left, write 1
move right 
move right, write 1 
move to uninitialized left, write 1
move to uninitialized right, write 1
move off end left
move off end right

**Comment by nevakrien** [comment]
This is a good start  for tests but I will integrate it into the existing testing. Just to be sure. 

 most of the work is already done in the feature branch I just need to set the linker in the cross compilation case to be the right linker. And to also compile io.c for arm In the cross compilation case.

**Pull request closed without merging**